tests/integration/targets/rds_instance_modify/tasks/main.yml

```diff
@@ -193,6 +193,63 @@
       - result.changed
       - result.db_instance_identifier == "{{ modified_instance_id }}"
 
+
+  # Test modifying CA certificate identifier -------------------------------------------
+
+  - name: Modify the CA certificate identifier to rds-ca-ecc384-g1 - check_mode
+    rds_instance:
+      state: present
+      db_instance_identifier: '{{ modified_instance_id }}'
+      allow_major_version_upgrade: true
+      ca_certificate_identifier: rds-ca-ecc384-g1
+      apply_immediately: true
+      tags:
+        Name: '{{ modified_instance_id }}'
+        Created_by: Ansible rds_instance tests
+    register: result
+    check_mode: true
+    vars:
+      ansible_python_interpreter: "{{ botocore_virtualenv_interpreter }}"
+
+  - name: Get curent CA certificate identifier
+    rds_instance_info:
+      db_instance_identifier: '{{ modified_instance_id }}'
+    register: db_info
+  - name: Assert that CA certificate identifier has been modified - check_mode
+    assert:
+      that:
+        - result is changed
+        - result is not failed
+        - db_info.instances[0].ca_certificate_identifier != "rds-ca-ecc384-g1"
+
+  - name: Modify the CA certificate identifier to rds-ca-ecc384-g1
+    rds_instance:
+      state: present
+      db_instance_identifier: '{{ modified_instance_id }}'
+      allow_major_version_upgrade: true
+      ca_certificate_identifier: rds-ca-ecc384-g1
+      apply_immediately: true
+      tags:
+        Name: '{{ modified_instance_id }}'
+        Created_by: Ansible rds_instance tests
+    register: result
+    vars:
+      ansible_python_interpreter: "{{ botocore_virtualenv_interpreter }}"
+
+  - name: Get curent CA certificate identifier
+    rds_instance_info:
+      db_instance_identifier: '{{ modified_instance_id }}'
+    register: db_info
+    retries: 20
```

Comment:
fyi, update operation takes a couple of minutes for change in caCertificateIdentifier.